apps/dashr-ncbi-explorer/app.R

```diff
@@ -0,0 +1,654 @@
+        height = 750,
+        width = 1150,
+        opacity = 0.5
+      )
+    }
+    return(alignment_chart)
+  }
+)
+
+#Callbacks for database search output.
+
+app$callback(
+  output(id = "search-output", property = "children"),
+  params = list(
+    input(id = "search-button", property = "n_clicks"),
+    state(id = "search-input", property = "value")
+  ),
+  
+  update_search_results <- function(n_clicks, search_term) {
+    if (n_clicks > 0) {
+      search_term <- gsub('"', "", search_term)
+      
+      search <- entrez_search(db = "nuccore", term = search_term, retmax=10)
+      
+      search_seqs <- entrez_fetch(db = "nuccore", id = search$ids, rettype = "fasta")
+      
+      red <- unlist(str_extract_all(search_seqs, ">.*\\n"))
+      
+      titles_vector <- c()
+      
+      for (i in red) {
+        title <- str_extract_all(i, ">.*?[:blank:]")
+        title <- substring(title, 2)
+        titles_vector <- c(titles_vector, title)
+      }
+      
+      sequence_titles <- as.list(unlist(str_extract_all(search_seqs, ">.*\\n")))
+      
+      sequence_dataframe <- do.call(rbind.data.frame, sequence_titles)
+      
+      names(sequence_dataframe)[1] <- "Dataset Accession ID and Description"
+      
+      sequence_dataframe$Accession_ID <- titles_vector
+      
+      sequence_dataframe <- sequence_dataframe[, c(2,1)]
+      
+      results_table <- dashDataTable(
+        id = "table",
+        style_table = list('overflowX' = 'scroll', 'overflowY' = 'scroll', 'maxHeight' = '100'),
+        style_data = list(
+          whiteSpace = "normal"
+        ),
+        css = list(
+          list(
+            selector = '.dash-cell div.dash-cell-value',
+            rule = 'display: inline; white-space: inherit; overflow: inherit; text-overflow: inherit;'
+          )
+        ),
+        columns = lapply(colnames(sequence_dataframe),
+                         function(colName){
+                           list("name" = colName,
+                                "id" = colName)
+                         }),
+        data = df_to_list(sequence_dataframe)
+        # ,
+        # 
+        # active_cell <- list('row' = 0, 'column' = 0, 'column_id' = 'Accession_ID')
+      )
+      
+      return(results_table)
+    }
+  }
+)
+
+app$callback(
+  output(id = "dataframe-store", property = "data"),
+  params = list(
+    input(id = "search-button", property = "n_clicks"),
+    state(id = "search-input", property = "value")
+  ),
+  
+  update_search_results <- function(n_clicks, search_term) {
+    if (n_clicks > 0) {
+      search_term <- gsub('"', "", search_term)
+      
+      search <- entrez_search(db = "nuccore", term = search_term, retmax=10)
+      
+      search_seqs <- entrez_fetch(db = "nuccore", id = search$ids, rettype = "fasta")
+      
+      red <- unlist(str_extract_all(search_seqs, ">.*\\n"))
+      
+      titles_vector <- c()
+      
+      for (i in red) {
```

Comment:
`lapply` ?

Reply:
For this I want the output to be a vector, so it's a little unnecessary to use lapply.